server/db.py

```
@@ -386,6 +387,38 @@ def update_one(conn, sql: str, parameters=()) -> bool:
+@contextmanager
+def writing(conn):
+    """רצף כתיבות שמסתיים ב-``commit`` — ובכל מסלול יציאה אחר ב-``rollback``.
+
+    זו ההשלמה של ``update_one`` לרצפים. כתיבה **בודדת** שנכשלת אינה
+    משאירה דבר: היא לא הספיקה לתפוס את נעילת הכתיבה. אבל ברצף, הכתיבה
+    הראשונה כבר תפסה אותה, ואם השנייה זורקת ואיש אינו עושה ``rollback``
+    — הנעילה נשארת יתומה. החיבור הוא של התהליכון (``Database``) והוא
+    נשאר בחיים, ולכן היא משוחררת רק בכתיבה המוצלחת הבאה *באותו
+    תהליכון*; בינתיים כל השאר ממתינים ``busy_timeout`` שלם ומקבלים
+    "database is locked" (#54, ‏#184, ‏#200, ‏#290).
+
```

Ruff (0.16.3)

[error] Contains control characters that can permit obfuscated code

(PLE2502)